Details row should update fix

XpRv opened the pull request:
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #10079
- [X] Include a change request file using `$ yarn change`

#### Description of changes

The negation of shallowCompare should be used to ensure that the component updates when a prop has changed. This change only affects DetailsList when useReducedRowRenderer is set to true.

Currently, this was set to false in all the DetailsList examples, however, with this additional bugfix, I have validated each of the examples are now working as expected. And that the render is occurring when the props are detected to have changed.

#### Focus areas to test

npm run build jest PASS
npm run buildfast PASS

For manual validation, I verified each of the examples Advanced, Animation, Basic, Compact, CustomFooter, CustomGroupHeaders, CustomRows, DragDrop, Grouped, NavigatingFocus. I inserted a console.log into the rend of DetailsRowBase to ensure it was occurring when expected and the performance fix is intact. For each example, I ran through all the expected functionality to ensure no regressions to the DetailsList occur when this feature is enabled.
(I can add my full test notes if desired, but in brief):
Validated:
Drag&Drop
Column re-sizing
Filtering
Selection (All, single, multiple)
Filtering
Animation
Scrolling (horizontal and vertical)
sticky footer
sticky header
Expand/Collapse all groups
Expand/Collapse single group
Customgroup headers
Navigating focus
Custom rows

The risk of this change is minimal for DetailsList users since it is behind this feature flag (flight) - which currently doesn't work as expected. The examples have all been validated for no regressions, so I have re-enabled this optimization.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10080)

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/OfficeDev/office-ui-fabric-react?pullRequest=10080) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/OfficeDev/office-ui-fabric-react?pullRequest=10080) before we can accept your contribution.<br/><br/>:x: XpRv [sign now](https://cla.opensource.microsoft.com/OfficeDev/office-ui-fabric-react?pullRequest=10080)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/OfficeDev/office-ui-fabric-react?pullRequest=10080) it.</sub>

KevinTCoughlin commented:
@XpRv, the fix makes sense. Thanks! It is a shame no test caught this..

One note, I think the changes to the examples should be reverted to _not use_ `useReducedRowRenderer`. A number of folks copy and paste the examples into their application and may be unaware of the trade-offs associated with using that prop.

Would you be open to reverting those changes?

XpRv commented:
> A number of folks copy and paste the examples into their application and may be unaware of the trade-offs associated with using that prop.

@KevinTCoughlin - thanks for the feedback, that's a good point! I reverted the changes to the examples and updated the change request file to reflect the latest iteration. (also re-ran the ut and fastbuild - PASS!)

KevinTCoughlin commented:
Thanks @XpRv. I'm amazed that this has remained uncaught since it was introduced in https://github.com/OfficeDev/office-ui-fabric-react/pull/5571.
